lib/AST/DocComment.cpp

```diff
@@ -426,24 +426,24 @@ class CommentProviderFinder final {
     return std::nullopt;
   }
 
+  /// Check if there is an inherited protocol that has a default implementation
+  /// of `VD` with a doc comment.
   std::optional<ResultWithDecl> findDefaultProvidedDecl(const ValueDecl *VD) {
-    // Only applies to protocol extension member.
-    auto *protocol = VD->getDeclContext()->getExtendedProtocolDecl();
-    if (!protocol)
+    NominalTypeDecl *nominalType =
+        dyn_cast_or_null<NominalTypeDecl>(VD->getDeclContext()->getAsDecl());
```

Comment:
What if `VD` is defined in an extension? Do we want to support that?

Reply:
That only handles protocol extensions, right? I was thinking of e.g a struct extension